pyriemann/datasets/simulated.py

```diff
@@ -222,3 +258,131 @@ def make_outliers(n_matrices, mean, sigma, outlier_coeff=10,
         outliers[i] = mean_sqrt @ powm(Oi, epsilon) @ mean_sqrt
 
     return outliers
+
+
+def make_classification_transfer(n_matrices, class_sep=3.0, class_disp=1.0,
+                                 domain_sep=5.0, theta=0.0, stretch=1.0,
+                                 random_state=None):
+    """Generate source and target toy datasets for transfer learning examples.
+
+    Generate a dataset with 2x2 SPD matrices drawn from two Riemannian Gaussian
+    distributions. The distributions have the same class dispersions and the
+    distance between their centers of mass is an input parameter. We can
+    stretch the target dataset and control a rotation matrix that maps the
+    source to the target domains. This function is useful for testing
+    classification or clustering methods on transfer learning applications.
+
+    Parameters
+    ----------
+    n_matrices : int, default=100
+        How many 2x2 matrices to generate for each class on each domain.
+    class_sep : float, default=3.0
+        Distance between the centers of the two classes.
+    class_disp : float, default=1.0
+        Dispersion of the data points to be sampled on each class.
+    domain_sep : float, default=5.0
+        Distance between the global means of each source and target datasets.
+    theta : float, default=0.0
+        Angle of the 2x2 rotation matrix from source to target dataset.
+    stretch : float, default=1.0
+        Factor to stretch the data points in target dataset. Note that when it
+        is != 1.0 the class dispersions in target domain will be different than
+        those in source domain (fixed at class_disp).
+    random_state : None | int | RandomState instance, default=None
+        Pass an int for reproducible output across multiple function calls.
+
+    Returns
+    -------
+    X_enc : ndarray, shape (4*n_matrices, 2, 2)
+        Set of SPD matrices.
+    y_enc : ndarray, shape (4*n_matrices,)
+        Extended labels for each data point.
+
+    Notes
+    -----
+    .. versionadded:: 0.3.1
+    """
+
+    rs = check_random_state(random_state)
+    seeds = rs.randint(100, size=4)
+
+    # the examples considered here are always for 2x2 matrices
+    n_dim = 2
+
+    # create a source dataset with two classes and global mean at identity
+    M1_source = np.eye(n_dim)  # first class mean at Identity at first
+    X1_source = sample_gaussian_spd(
+        n_matrices=n_matrices,
+        mean=M1_source,
+        sigma=class_disp,
+        random_state=seeds[0])
+    y1_source = (1*np.ones(n_matrices)).astype(int)
```

Comment:
Names of classes should be defined by a parameter, with default value `class_names = [1, 2]`.